packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackinstances-action.ts

```diff
@@ -0,0 +1,97 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import { Action } from '../action';
+import { validatePercentage } from '../common';
+import { SingletonPolicy } from './private/singleton-policy';
+import { CommonCloudFormationStackSetOptions, StackInstances, StackSetParameters } from './stackset-types';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+ * Properties for the CloudFormationDeployStackInstancesAction
+ */
+export interface CloudFormationStackInstancesActionProps extends codepipeline.CommonAwsActionProps, CommonCloudFormationStackSetOptions {
+  /**
+   * The name of the StackSet we are adding instances to
+   */
+  readonly stackSetName: string;
+
+  /**
+   * Specify where to create or update Stack Instances
+   *
+   * You can specify either AWS Accounts Ids or AWS Organizations Organizational Units.
+   */
+  readonly stackInstances: StackInstances;
+
+  /**
+   * Parameter values that only apply to the current Stack Instances
+   *
+   * These parameters are shared between all instances added by this action.
+   *
+   * @default - no parameters will be overridden
+   */
+  readonly parameterOverrides?: StackSetParameters;
+}
+
+/**
+ * CodePipeline action to create/update Stack Instances of a StackSet
+ *
+ * After the initial creation of a stack set, you can add new stack instances by
+ * using CloudFormationStackInstances. Template parameter values can be
+ * overridden at the stack instance level during create or update stack set
+ * instance operations.
+ *
+ * Each stack set has one template and set of template parameters. When you
+ * update the template or template parameters, you update them for the entire
+ * set. Then all instance statuses are set to OUTDATED until the changes are
+ * deployed to that instance.
+ */
+export class CloudFormationDeployStackInstancesAction extends Action {
```

Comment:
I still think this action should contain the word "StackSet" in it, otherwise it's a little misleading.

I would ask for it to be renamed to `CloudFormationDeployStackSetInstancesAction`. I understand you hate long names, but I think those 3 characters here are really worth it.